worksheet_D/oxo.py

```diff
@@ -0,0 +1,121 @@
+class OxoBoard:
+    def __init__(self):
+        self.board = {(0,0):0,(0,1):0,(0,2):0,(1,0):0,(2,0):0,(1,1):0,(2,2):0,(1,2):0,(2,1):0}
```

Comment:
The order of initialisation here is a bit odd -- would be easier to verify that it's correct if they were in e.g. left-to-right, top-to-bottom order.

Even better woudl be initialising the dictionary in a loop, e.g.
```
self.board = {}
for x in xrange(3):
    for y in xrange(3):
        self.board[x, y] = 0
```